### DISCUSSION TITLE
ProjectCategory matching query does not exist

### DISCUSSION THREAD

**Discussion by bnouyou** [broadinstitute/seqr#2002] [Q&A] [answered]
Hello,

I have a question about the utilization of seqr. 
In the variants search page, There is a "seqr" link under the variant information. As I have understood, it should bring to a page which displays this variant found on other samples loaded in seqr. Am I right ?
![image](https://user-images.githubusercontent.com/22322228/127458380-52899056-3b97-49e0-aea5-0dd4df7d96d2.png)

And now, when I click this lin, I have this error message:
![image](https://user-images.githubusercontent.com/22322228/127458618-edad8c56-8eff-4028-b73e-e4560de7ecd8.png)

What is Project Category ?

Thank you for your help.

**Comment by hanars** [comment]
can you include the seqr server logs? Its hard to debug without more information

**Reply by bnouyou** [reply]
This is the logs when I click on "seqr" link:
```
seqr_1             | {"timestamp": "2021-07-29 15:33:48,639", "severity": "INFO", "httpRequest": {"requestMethod": "GET", "requestUrl": "http://lx165/variant_search/results/45f7bb0f97085f502699688dd272148c", "status": 200, "responseSize": 1216, "userAgent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36", "remoteIp": "172.18.8.26", "referer": "http://lx165/variant_search/results/22b5c38ec168be79745ac5a30b69f98f?page=1&sort=xpos", "protocol": "HTTP/1.1"}, "user": "benedicte.nouyou@chu-rennes.fr"}
seqr_1             | {"timestamp": "2021-07-29 15:33:49,387", "severity": "INFO", "httpRequest": {"requestMethod": "GET", "requestUrl": "http://lx165/api/saved_search/all", "status": 200, "responseSize": 17798, "userAgent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36", "remoteIp": "172.18.8.26", "referer": "http://lx165/variant_search/results/45f7bb0f97085f502699688dd272148c", "protocol": "HTTP/1.1"}, "user": "benedicte.nouyou@chu-rennes.fr"}
seqr_1             | {"timestamp": "2021-07-29 15:33:49,599", "severity": "WARNING", "httpRequest": {"requestMethod": "POST", "requestUrl": "http://lx165/api/search/45f7bb0f97085f502699688dd272148c?sort=xpos&page=1", "status": 400, "responseSize": 65, "userAgent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36", "remoteIp": "172.18.8.26", "referer": "http://lx165/variant_search/results/45f7bb0f97085f502699688dd272148c?page=1&sort=xpos", "protocol": "HTTP/1.1"}, "requestBody": {"allProjectFamilies": true, "search": {"locus": {"rawVariantItems": "9-138586966-GC-AA", "genomeVersion": "37"}}}, "message": "ProjectCategory matching query does not exist.", "user": "benedicte.nouyou@chu-rennes.fr"}
seqr_1             | {"timestamp": "2021-07-29 15:33:49,599", "severity": "WARNING", "message": "ProjectCategory matching query does not exist.: /api/search/45f7bb0f97085f502699688dd272148c"}
seqr_1             | {"timestamp": "2021-07-29 15:33:49,736", "severity": "INFO", "httpRequest": {"requestMethod": "POST", "requestUrl": "http://lx165/api/search_context", "status": 200, "responseSize": 17992, "userAgent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36", "remoteIp": "172.18.8.26", "referer": "http://lx165/variant_search/results/45f7bb0f97085f502699688dd272148c?page=1&sort=xpos", "protocol": "HTTP/1.1"}, "requestBody": {"searchHash": "45f7bb0f97085f502699688dd272148c", "searchParams": {"allProjectFamilies": true, "search": {"locus": {"rawVariantItems": "9-138586966-GC-AA", "genomeVersion": "37"}}}}, "user": "benedicte.nouyou@chu-rennes.fr"}

```

**Comment by hanars** [comment]
oh this is actually a bug in seqr, I'm sorry. As a workaround, you can create a ProjectCategory in the django admin named "Demo" and this will be resolved, but I'll try to get a real fix up next week

**Reply by hanars** [reply]
https://github.com/broadinstitute/seqr/issues/2007 to track the fix

**Reply by hanars** [reply]
Update: the code is fixed in the latest version of master so this will no longer through an error if you don't have a Demo category

**Reply by bnouyou** [reply]
Thank you !

I haven't dowloaded the lastest version yet, but I have just tried the workaround with Demo category. And now I don't have the same error message. In the browser this is the message:

![image](https://user-images.githubusercontent.com/22322228/128684063-9dc0767c-db91-4b99-a2ba-3fcd7b2e4dae.png)

And this is the logs:
```
seqr_1             | {"timestamp": "2021-08-09 09:06:26,111", "severity": "WARNING", "httpRequest": {"requestMethod": "POST", "requestUrl": "http://lx165/api/search/c770b34abe9d8961f64642564be168f9?sort=xpos&page=1", "status": 400, "responseSize": 1246, "userAgent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36", "remoteIp": "172.18.8.26", "referer": "http://lx165/variant_search/results/c770b34abe9d8961f64642564be168f9?page=1&sort=xpos", "protocol": "HTTP/1.1"}, "requestBody": {"allProjectFamilies": true, "search": {"locus": {"rawVariantItems": "6-108985337-T-C", "genomeVersion": "37"}}}, "message": "No es index found for families ", "user": "benedicte.nouyou@chu-rennes.fr", "traceback": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py\", line 181, in _get_response\n    response = wrapped_callback(request, *callback_args, **callback_kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/django/contrib/auth/decorators.py\", line 21, in _wrapped_view\n    return view_func(request, *args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/django/contrib/auth/decorators.py\", line 21, in _wrapped_view\n    return view_func(request, *args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/django/contrib/auth/decorators.py\", line 21, in _wrapped_view\n    return view_func(request, *args, **kwargs)\n  File \"/seqr/seqr/views/apis/variant_search_api.py\", line 70, in query_variants_handler\n    skip_genotype_filter=is_all_project_search, user=request.user)\n  File \"/seqr/seqr/utils/elasticsearch/utils.py\", line 113, in get_es_variants\n    user=user,\n  File \"/seqr/seqr/utils/elasticsearch/es_search.py\", line 44, in __init__\n    ', '.join([f.family_id for f in families])))\nseqr.utils.elasticsearch.utils.InvalidSearchException: No es index found for families \n"}
seqr_1             | {"timestamp": "2021-08-09 09:06:26,111", "severity": "WARNING", "message": "Bad Request: /api/search/c770b34abe9d8961f64642564be168f9"}
```


Will it be solved in latest version ?

Thank you for your help.

**Reply by hanars** [reply]
it means your user does not have access to any seqr projects with elasticsearch indices added to them

**Reply by bnouyou** [reply]
How can I solve this problem ? Actually my user is SuperUser and Staff User. This is the user I use for managing eveything is my seqr instance. 
Connected with this user, I can access to all my seqr projects and all samples.

**Reply by hanars** [reply]
To clarify, did you add any projects to the demo project category? Because if so they wouldn't be included in the search and that might explain this issue

**Reply by bnouyou** [reply]
I have included my projects in the Demo category. I shouldn't do that ?

**Comment by hanars** [comment, marked as answer]
Demo projects mean "not real data" and are excluded from the cross project
search. So if you want projects to be in that search you need to NOT put
them in the demo category. alternatively, you can pull the latest version
of master and delete the Demo category entirely

**Reply by bnouyou** [reply]
So I have created a Demo category, but I should not add any project in that category ? So my projects should not have any category, right ? But in that case I still have the error message I have at first.
Sorry I don't really understand how work the workaround. I think I will update the lastest version asap.

**Reply by hanars** [reply]
If you remove all the projects from a category in the seqr UI it deletes the category thus bringing you back to where you started. If you want to create the demo category, you need to create it in the django admin with no projects in it, which is the only way to create an empty category in seqr, you can't do it via the UI. But I agree updating the code is better

**Reply by bnouyou** [reply]
Ok I see. Actually I can't create a project category in Django Admin and not assigning it to any project. I assigned it to only one project, and now it's working.
I think I will update the code asap, it will be better. But for now, it's working.
Thank you for your help.
